crates/codegen/parser/runtime/src/cursor.rs

```diff
@@ -64,7 +64,7 @@ pub struct Cursor {
     /// The list of ancestor rule nodes that the `current` node is a part of.
     path: Vec<PathRuleNode>,
     /// The node the cursor is currently pointing to.
-    current: PathNode,
+    pub(super) current: PathNode,
```

Comment:
I wonder if the only usage for this is the one in `matches_query_node_id()` .. If so, what do you think of adding a `Cursor::node_ref()` similar to `Cursor::node()` that returns a `&Node` instead of cloning .. I imagine it is useful for the public API even .. as many user operations don't require cloning .. instead of exposing these implementation details (`Cursor::current` and `PathNode`) ..

Reply:
I think that an accessors such as `node()` should return a reference, and not clone a value. That is contrary to my expectation about accessors. However I want to talk about that with the three of us, so for now I will use the accessor and pretend it is a reference.